### PR TITLE
store an in-memory history of the last addresses

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8b67710022d854ba945909135fb10c5b86bf1223 available at: https://gist.github.com/2de5dd315dff40192af5<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5091.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5091.dmg

File: libraries/networking/src/AddressManager.cpp [hunk 1]
```diff
@@ -500,3 +509,22 @@ void AddressManager::copyAddress() {
...
+void AddressManager::addCurrentAddressToHistory() {
+    if (_lastHistoryAppend == 0) {
+        // we don't store the first address on application load
+        // just update the last append time so the next is stored
+        _lastHistoryAppend = usecTimestampNow();
+    } else {
+        const quint64 DOUBLE_STORE_THRESHOLD_USECS = 500000;
+
+        // avoid double storing when the host changes and the viewpoint changes immediately after
+        if (usecTimestampNow() - _lastHistoryAppend > DOUBLE_STORE_THRESHOLD_USECS) {
```
Comment (Atlante45) [inline libraries/networking/src/AddressManager.cpp]: Checking `currentAddress()` against `_history.last()` might be more reliable.
Plus it'll prevent a bit of redundancy if the user keeps moving around then jumping back to the same place.

Comment (birarda) [inline libraries/networking/src/AddressManager.cpp]: Problem is, this guards against a double store where you've just changed
hosts, and then the path changes (after a lookup).

So the thing you're trying to double store doesn't match what you just put
in the history at all.

Comment (Atlante45) [inline libraries/networking/src/AddressManager.cpp]: Oh gotcha

**Pull request merged**